GitCommands/Git/Executable.cs

```diff
@@ -31,21 +32,50 @@ public Executable([NotNull] Func<string> fileNameProvider, [NotNull] string work
 
         /// <inheritdoc />
         [PermissionSet(SecurityAction.Demand, Name = "FullTrust")]
-        public IProcess Start(ArgumentString arguments = default, bool createWindow = false, bool redirectInput = false, bool redirectOutput = false, Encoding outputEncoding = null)
+        public IProcess Start(ArgumentString arguments = default, bool createWindow = false, bool redirectInput = false, bool redirectOutput = false, Encoding outputEncoding = null, bool useShellExecute = false)
         {
-            // TODO should we set these on the child process only?
-            EnvironmentConfiguration.SetEnvironmentVariables();
+            string fileName = "<unknown executable>";
+            string args = null;
 
-            var args = (arguments.Arguments ?? "").Replace("$QUOTE$", "\\\"");
+            return HandleExceptions(
+                () =>
+                {
+                    fileName = _fileNameProvider();
+
+                    args = (arguments.Arguments ?? "").Replace("$QUOTE$", "\\\"");
 
-            var fileName = _fileNameProvider();
+                    // TODO should we set these on the child process only?
+                    EnvironmentConfiguration.SetEnvironmentVariables();
 
-            return new ProcessWrapper(fileName, args, _workingDir, createWindow, redirectInput, redirectOutput, outputEncoding);
+                    return new ProcessWrapper(fileName, args, _workingDir, createWindow, redirectInput, redirectOutput, outputEncoding, useShellExecute);
+                },
+                () => new ProcessStartInfo
+                {
+                    FileName = fileName,
+                    Arguments = args,
+                    WorkingDirectory = _workingDir
+                });
```

Comment:
Why do we need all this plumbing? The exception is caught and then handled in `ProcessWrapper`?

Reply:
> Why do we need all this plumbing? The exception is caught and then handled in `ProcessWrapper`?

The exception is not consumed. It is wrapped into `ExecutableException`.

One could consider to replace the arguments `fileName, args, _workingDir` with a prefilled `ProcessStartInfo`.
Though the extension method `GetOutput` below would have to be adapted, too.